mjidris/irondb
# Us134

## Objective
Remove all backend code from the frontend /public/javascripts/register.js and places it correctly into the backend controller. 

## Detailed Description
I also took the opportunity to fix the bootstrap layout including but not limited to properly centering the content on all sizes of browser windows. Both username and email redundancy checks are now done asynchronously 500ms after the user stops typing in the textbox (to prevent spamming the controller) and if there is a duplicate the user is notified instantly without preloading all usernames into the page. In order to achieve this I created an endpoint /register/user/[username] that returns a boolean for if the username exists. The frontend now only checks that boolean instead of a list of all usernames.

## Screenshot (if applicable)
![image](https://user-images.githubusercontent.com/48575921/69120637-12697880-0a68-11ea-9513-2267c625a8b5.png)
![image](https://user-images.githubusercontent.com/48575921/69120647-1bf2e080-0a68-11ea-96f3-96b1937056ee.png)


## How to Test (if applicable)
1. Create a new user and attempt to use existing username/email
2. View source and verify that all usernames are no longer being spit out on page

## Tests (if added)
N/A

## Requirements
[Taiga User Story](https://tree.taiga.io/project/cbrobsto-iron-db/us/134)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/138)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/148)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/147)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/136)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/135)
